packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -92,6 +93,9 @@ export default class ChildProcessWorker implements WorkerInterface {
       } as NodeJS.ProcessEnv,
       // Suppress --debug / --inspect flags while preserving others (like --harmony).
       execArgv: process.execArgv.filter(v => !/^--(debug|inspect)/.test(v)),
+      // default to advanced serialization in order to match worker threads
+      // @ts-expect-error
+      serialization: 'advanced',
```

Comment:
I'm not sure if we should change this by default.

* it isn't supported by all node versions
* The passed data might differ from what is passed via JSON. 

> However, this format is not a full superset of JSON, and e.g. properties set on objects of such built-in types will not be passed on through the serialization step. Additionally, performance may not be equivalent to that of JSON, depending on the structure of the passed data. Therefore, this feature requires opting in by setting the serialization option to 'advanced' when calling child_process.spawn() or child_process.fork(). [source](https://nodejs.org/api/child_process.html#child_process_advanced_serialization)

I suggest that we allow callers to customize that setting (already possible today with `forkOptions`.

Reply:
the reason I changed it is that it then matches `worker_threads`. People can still customize it as you point out, but our internal defaults is then more similar.

Can of course do that separately to not conflate the change with the bug fix

Reply:
Could be worth to do so or at least call mention it explicitly as a breaking change in the changelog.

Reply:
extracted to #10983 and removed from this PR